service/raft/raft_group0_client.cc

```diff
@@ -239,6 +239,10 @@ future<group0_guard> raft_group0_client::start_operation(seastar::abort_source*
         on_internal_error(logger, "start_group0_operation: must run on shard 0");
     }
 
+    if (_maintenance_mode) {
+        throw exceptions::configuration_exception{"cannot start group0 operation in the maintenance mode"};
```

Comment:
Why do you need this? If we already pretend to be in the recovery mode the switch below will use pre-raft procedure.

Reply:
Yes, and we don't want to use it. The user would be able to change the schema, resulting in schema inconsistency (it's changed locally and not propagated to other nodes)

Reply:
I am not sure you want to prevent this. The whole idea of this mode is that it is off limits.